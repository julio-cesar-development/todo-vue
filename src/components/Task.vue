<template>
  <div class="task" v-bind:class="stateClass" v-on:click.stop="changeTaskState()">
    <span class="close" v-on:click.stop="deleteTask()">X</span>
    <span>
      {{ task.name }}
    </span>
  </div>
</template>

<script>
export default {
  name: 'Task',
  props: {
    task: {
      type: Object,
      required: true,
    }
  },
  computed: {
    stateClass() {
      return {
        pending: this.task.pending,
        done: !this.task.pending,
      }
    },
  },
  methods: {
    deleteTask() {
      this.$emit('task-deleted', this.task);
    },
    changeTaskState() {
      this.$emit('task-state-changed', this.task);
    },
  },
}
</script>

<style scoped>
.task {
  box-sizing: border-box;
  width: 280px;
  height: 100px;
  padding: 10px;
  border-radius: 0.3em;
  font-size: 1.2em;
  font-weight: 300;
  cursor: pointer;
  user-select: none;
  display: flex;
  justify-content: center;
  align-items: center;
  position: relative;
  transition: .5s;
}

.pending {
  border-left: 8px solid #b73228;
  background-color: #f44336;
}

.done {
  color: #ddd;
  border-left: 8px solid #0a8f08;
  background-color: #4caf50;
  text-decoration: line-through;
}

.close {
  box-sizing: border-box;
  position: absolute;
  top: 10px;
  right: 10px;
  border-radius: 50%;
  height: 20px;
  width: 20px;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 0.8em;
  font-weight: bold;
  cursor: pointer;
  transition: .5s;
}

.pending .close {
  background-color: #b73229;
}

.done .close {
  background-color: #0a8f08;
}
</style>
