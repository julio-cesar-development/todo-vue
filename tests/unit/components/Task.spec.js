import { shallowMount } from '@vue/test-utils';
import Task from '@/components/Task.vue';

describe('Task.vue', () => {
  it('Task.vue should match all specifications bellow', () => {
    const props = {
      task: {
        name: 'Test 01',
        pending: false,
      },
    };
    const wrapper = shallowMount(Task, {
      'propsData': props,
    });
    expect(wrapper.isVueInstance()).toBeTruthy();

    expect(wrapper.classes('done')).toBe(true);

    expect(wrapper.contains('div[class*="task"]')).toBe(true);
    expect(wrapper.contains('label[class="task-label"]')).toBe(true);

    const labelTask = wrapper.find('.task-label');
    expect(labelTask.html()).toEqual(`<label class="task-label">${props.task.name}</label>`);

    expect(wrapper).toMatchSnapshot();
  });
});