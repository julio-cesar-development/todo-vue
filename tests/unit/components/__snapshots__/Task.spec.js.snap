// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task.vue Task.vue should match all specifications bellow 1`] = `
<div class="task done"><span class="icon-rounded icon-rounded--left">âœ“
  </span> <span class="icon-rounded icon-rounded--right">X
  </span> <label class="task-label">Test 01</label> <input class="task-input" style="display: none;"></div>
`;
